.github/workflows/node.js.yml

```
@@ -20,7 +20,46 @@
         with:
           node-version: ${{ matrix.node-version }}
-          cache: 'npm'
+
+      # Clear npm cache
+      - name: Clear npm cache
+        run: npm cache clean --force
+
+      # Check if package.json has changed
+      - name: Check if package.json has changed
+        id: check_packages
+        run: |
```

actionlint

shellcheck reported issue in this script: SC2086:info:1:85: Double quote to prevent globbing and word splitting

(shellcheck)